shoes-core/lib/shoes/download.rb

```diff
@@ -140,9 +142,23 @@ def save_to_file(file_path)
       open(file_path, 'wb') { |fw| fw.print @response.body }
     end
 
-    def download_started(content_length)
-      @content_length = content_length
-      @started        = true
+    def download_started(response)
+      unless @started
+        @started = true
+        @content_length = read_content_length(response)
+
+        @response.status = [response.code, response.message]
```

Comment:
Yup, it's about compatibility. It was leaning on the [`OpenURI::Meta`](https://ruby-doc.org/stdlib-2.3.0/libdoc/open-uri/rdoc/OpenURI/Meta.html) object which has this odd form.